proxy/ptunnel/commands/agent.go

```diff
@@ -4,47 +4,90 @@ import (
+	c.AddCommand(NewListAgentCommand(bf))
+	return c
+}
+func NewListAgentCommand(bf *client.BaseFlags) *cobra.Command {
+	c := &cobra.Command{
+		Use: "list",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			agent, err := kcerts.FindSSHAgent(bf.Local, bf.Log)
+			if err != nil {
+				return err
+			}
+			principals, err := agent.Principals()
+			if err != nil {
+				return err
+			}
+			for _, p := range principals {
+				fmt.Printf("PKS: %s Identities: %v ValidFor: %s \n", p.MD5, p.Principals, p.ValidFor.String())
+			}
+			return nil
+		},
+	}
+	return c
 }
 
-func NewAgentCommand(bf *client.BaseFlags) *cobra.Command {
-	agentConfig := &agentConfig{}
+func NewRunAgentCommand(parent *cobra.Command, bf *client.BaseFlags) *cobra.Command {
 	c := &cobra.Command{
+		Use:   "run -- [COMMAND]",
+		Short: "Runs the following command using the enkit ssh-agent",
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return RunAgentCommand(bf, agentConfig)
+			return RunAgentCommand(parent, bf, args)
 		},
-		Use:   "agent",
-		Short: "commands for the enkit specific ssh-agent",
 	}
-	// Note the following is intended to be user friendly, identities here are cert principals
-	c.Flags().BoolVarP(&agentConfig.ListIdentities, "list-identities", "l", false, "list the identities loaded current in the agent")
-	c.Flags().BoolVarP(&agentConfig.Print, "print", "p", false, "print the socket and PID of the running agent")
 	return c
 }
 
-func RunAgentCommand(bf *client.BaseFlags, config *agentConfig) error {
+func RunAgentCommand(command *cobra.Command, bf *client.BaseFlags, args []string) error {
 	agent, err := kcerts.FindSSHAgent(bf.Local, bf.Log)
 	if err != nil {
 		return err
 	}
-	if config.Print {
-		fmt.Printf("The enkit agent is running at socket %s \n", agent.Socket)
-		fmt.Printf("The enkit agent's pid is %d \n", agent.PID)
-
+	shell := os.Getenv("SHELL")
+	if shell == "" {
+		shell = "sh"
 	}
-	if config.ListIdentities {
-		principals, err := agent.Principals()
-		if err != nil {
-			return err
-		}
-		for _, p := range principals {
-			fmt.Printf("PKS: %s Identities: %v ValidFor: %s \n", p.MD5, p.Principals, p.ValidFor.String())
+	var kwargs []string
+	if len(args) > 0 {
+		kwargs = append([]string{"-c"}, strings.Join(args[:], " "))
+	}
+	cmd := exec.Command(shell, kwargs...)
+	cmd.Stdout = command.OutOrStdout()
+	cmd.Stderr = command.ErrOrStderr()
+	cmd.Stdin = command.InOrStdin()
+	cmd.Env = append(os.Environ(), agent.GetEnv()...)
+	if err := cmd.Run(); err != nil {
+		if exit, ok := err.(*exec.ExitError); ok {
+			return kflags.NewStatusError(exit.ExitCode(), err)
 		}
+		return err
 	}
 	return nil
 }
+
+func NewPrintCommand(parent *cobra.Command, bf *client.BaseFlags) *cobra.Command {
+	c := &cobra.Command{
+		Use:   "print",
+		Short: "alias for agent run -- ssh-add -s",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return RunAgentCommand(parent, bf, []string{"ssh-agent", "-s"})
```

Comment:
uhm .. this creates a NEW ssh-agent. It does not print out the details of the ssh-agent created by enkit?

Reply:
Just using print now